Sources/Datadog/Tracer.swift

```diff
@@ -45,20 +45,25 @@ public struct DDTags {
 public typealias DDTracer = Tracer
 
 public class Tracer: OTTracer {
-    /// Builds the `Span` from user input.
-    internal let spanBuilder: SpanEventBuilder
-    /// Writes the `Span` to file.
-    internal let spanOutput: SpanOutput
+    internal let core: DatadogCoreProtocol
+    /// Span events mapper configured by the user, `nil` if not set.
+    internal let spanEventMapper: SpanEventMapper?
     /// Queue ensuring thread-safety of the `Tracer` and `DDSpan` operations.
     internal let queue: DispatchQueue
     /// Integration with RUM Context. `nil` if disabled for this Tracer or if the RUM feature is disabled.
     internal let rumContextIntegration: TracingWithRUMContextIntegration?
     /// Integration with Logging. `nil` if the Logging feature is disabled.
     internal let loggingIntegration: TracingWithLoggingIntegration?
 
-    private let dateProvider: DateProvider
     private let tracingUUIDGenerator: TracingUUIDGenerator
 
+    /// The service name that will appear in traces (if not provided or `nil`, the SDK default `serviceName` will be used).
+    internal let serviceName: String?
+    /// Enriches traces with network connection info.
+    /// This means: reachability status, connection type, mobile carrier name and many more will be added to every span and span logs.
+    /// For full list of network info attributes see `NetworkConnectionInfo` and `CarrierInfo`.
+    /// - Parameter enabled: `false` by default
+    internal let sendNetworkInfo: Bool
```

Comment:
All these (`spanEventMapper`, `serviceName` and `sendNetworkInfo`) are passed through `tracerConfiguration`. How about capturing the entire immutable `tracerConfiguration` on a property instead of copying its fields into `Tracer`? The main benefit would be maintainability - adding a new field will require a change only in a single place. This can be even a convention to both `Logger`, `Tracer` and `RUMMonitor`, making things easier - both in code, tests and mocks.